src/main/java/com/google/devtools/build/lib/rules/java/JavaCompileAction.java

```diff
@@ -312,7 +313,9 @@ private JavaSpawn getReducedSpawn(
     // These flags instruct JavaBuilder that this is a compilation with a reduced classpath and
     // that it should report a special value back if a compilation error occurs that suggests
     // retrying with the full classpath.
-    classpathLine.add("--reduce_classpath_mode", fallback ? "BAZEL_FALLBACK" : "BAZEL_REDUCED");
+    classpathLine.add("--reduce_classpath_mode", 
+        fallback ? "BAZEL_FALLBACK" 
+            : (classpathMode == JavaClasspathMode.BAZEL ? "BAZEL_REDUCED" : "BAZEL_REDUCED_NO_FALLBACK"));
```

Comment:
Instead of adding the new `--reduce_classpath_mode=BAZEL_REDUCED_NO_FALLBACK` mode, did you consider having Bazel invoke JavaBuilder with a reduced classpath and pass `--reduce_classpath_mode=NONE`, and then handle not doing fallback on the Bazel side? I think that would provide equivalent behaviour without JavaBuilder changes.

Reply:
Thx, that's much better. Pushed the change.

There is still 1 test failure, but seems unrelated to the change:
```
test_size_less_than_385MB FAILED: terminated because this command returned a non-zero status:
/private/var/tmp/_bazel_buildkite/1198ee8566c303641c6d358da6949c16/sandbox/darwin-sandbox/4799/execroot/_main/bazel-out/darwin_arm64-fastbuild/bin/src/test/shell/integration/minimal_jdk_test.runfiles/_main/src/test/shell/integration/minimal_jdk_test:50: in call to test_size_less_than_385MB

```